internal/services/r2_bucket/model.go

```diff
@@ -15,16 +14,8 @@ type R2BucketModel struct {
 	ID           types.String `tfsdk:"id" json:"-,computed"`
 	Name         types.String `tfsdk:"name" json:"name,required"`
 	AccountID    types.String `tfsdk:"account_id" path:"account_id,required"`
-	Location     types.String `tfsdk:"location" json:"locationHint,optional"`
+	Location     types.String `tfsdk:"location" json:"location,computed_optional"`
 	Jurisdiction types.String `tfsdk:"jurisdiction" json:"-,computed_optional"`
-	StorageClass types.String `tfsdk:"storage_class" json:"storageClass,computed_optional,no_refresh"`
+	StorageClass types.String `tfsdk:"storage_class" json:"storage_class,computed_optional"`
 	CreationDate types.String `tfsdk:"creation_date" json:"creation_date,computed"`
 }
-
-func (m R2BucketModel) MarshalJSON() (data []byte, err error) {
```

Comment:
why are we not using the marshaler? this is how state and diffs are calculated correctly based on the struct fields in the request/response and tags.

Reply:
The model's json field tags don't match the param names used in the Create Bucket request body (since I've aligned them to the response property names instead). Instead of manually marshaling JSON and passing a request body override, I'm leaving it up to the client SDK serialize. Would you rather I add a `CreateR2BucketModel` with json tags that align with the request params and use the internal marshaler instead?